domains/client/domain-operator/src/domain_block_processor.rs

```diff
@@ -788,7 +820,18 @@ where
                             "Failed to generate fraud proof: {err}"
                         )))
                     })?,
-                _ => unimplemented!("fraud proof not implemented for the invariants"),
+                ReceiptMismatchInfo::TotalRewardsMismatch { .. } => self
```

Comment:
We may not need the `TotalRewardsMismatch::consensus_block_hash` field since it is not used here.

Reply:
We still need it `find_first_unconfirmed_bad_receipt_info` though.